src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,157 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int INIT_SIZE = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    MyLinkedList() {
+        size = INIT_SIZE;
+    }
+
     @Override
     public void add(T value) {
+        if (size != 0) {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+            return;
+        }
+        initFirstNode(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            checkIndex(index);
```

Comment:
```suggestion
        if (index == size) {
        add(value);
        return
        }
        checkIndex(index);
```

nested if-else constructs are pretty hard to read, let's simplify a little